src/unlabel.ml

```diff
@@ -0,0 +1,192 @@
+    Compenv.module_of_filename fmt
+      labelled_mli_filename
+      labelled_mli_filename
+    |> Str.(replace_first (regexp "^CC") "")
+  in
+  let unlabelled_name = Str.(replace_first (regexp "Labels") "" labelled_name) in
+  let mli_text = slurp_file labelled_mli_filename in
+  let lexbuf = Lexing.from_string mli_text in
+  Location.init lexbuf labelled_mli_filename;
+  let labelled_ast = Parse.interface lexbuf in
+  let labelled_functions = ref [] in
+  let rec iterator =
+    let open Ast_iterator in
+    let open Parsetree in
+    { Ast_iterator.default_iterator with
+        value_description = fun iterator vdesc ->
+          match vdesc with
+            | { pval_name = { txt = name }
+              ; pval_type
+              ; pval_prim
+              ; pval_attributes
+              ; pval_loc
+              } ->
+              let rec loop = function
+                | { ptyp_desc = Ptyp_arrow (Labelled label, left, right)
+                  ; ptyp_loc = { loc_start = pos } }
+                  when
+                    List.for_all
+                      (fun ({Asttypes.txt}, _) -> txt <> "keep_label")
+                      left.ptyp_attributes
+                  ->
+                  iterator.typ iterator left;
+                  (label, pos) :: loop right
+                | { ptyp_desc = Ptyp_arrow (_, left, right) } ->
+                  iterator.typ iterator left;
+                  loop right
+                | typ ->
+                  iterator.typ iterator typ;
+                  []
+              in
+              let labels = loop pval_type in
+              if labels <> [] then
+                labelled_functions := (name, labels, pval_loc) :: !labelled_functions
+    }
+  in
+  iterator.signature iterator labelled_ast;
+  let labelled_functions = List.rev !labelled_functions in
+
+  Format.pp_print_flush fmt ();
+  Format.pp_open_vbox fmt 0;
+
+  (* Log all found labels *)
+  (*
+  List.iter
+    begin fun (name, labels, loc) ->
+      Format.pp_open_box fmt 2;
+      Location.print_loc fmt loc;
+      Format.fprintf fmt "@{<loc>:@}@,";
+      Format.pp_open_vbox fmt 4;
+      Format.fprintf fmt "Functional value \"%s\" has labels:" name;
+      List.iter
+        begin fun (label, {Lexing.pos_lnum; pos_bol; pos_cnum} ) ->
+          Format.fprintf fmt "@,(line%5i char%3i) %s"
+            pos_lnum (pos_cnum - pos_bol) label
+        end
+        labels;
+      Format.pp_close_box fmt ();
+      Format.pp_close_box fmt ();
+      Format.pp_print_cut fmt ();
+    end
+    labelled_functions;
+  *)
+
+  (* Create list containing only labels and postions, but no function names. *)
+  let labels =
+    List.map (fun (_,labels,_) -> labels) labelled_functions
+    |> List.flatten
+  in
+
+  begin (* check whether positions are in order *)
+    List.map (fun (_, {Lexing.pos_cnum}) -> pos_cnum) labels
+    |> List.fold_left (fun a b -> assert (a <= b); b) 0
+    |> ignore
+  end;
+
+  Format.pp_open_vbox fmt 0;
```

Comment:
That's pretty weird. I think it'd be better to use a single `Format.fprintf fmt "@[<v>…@]@." with sub-printers, and avoid the `\n` in format strings (use `@,` for inserting breaks instead).

Reply:
Removed the ``Format.`` stuff altogether. It was only needed while implementing the rewriter.